src/extension.ts

```diff
@@ -1279,9 +1283,12 @@ async function describeKubernetes(explorerNode?: ClusterExplorerResourceNode) {
             const nsarg = explorerNode.namespace ? `--namespace ${explorerNode.namespace}` : '';
             return { cmd: `describe ${explorerNode.kindName} ${nsarg}`, resourceKindName: explorerNode.kindName };
         } else {
-            const value = await findKindNameOrPrompt(kuberesources.commonKinds, 'describe', { nameOptional: true });
-            if (value) {
-                return { cmd: `describe ${value}`, resourceKindName: value };
+            const clusterKinds = await kubectlUtils.clusterResources(kubectl, ['get', 'list']);
+            if (clusterKinds.succeeded) {
+                const value = await findKindNameOrPrompt(clusterKinds.result, 'describe', { nameOptional: true, skipPrompt: true });
```

Comment:
I'm not sure users always want to wait for us to load a list of resource kinds followed by a list of resources.  With the current workflow, if I know which resource I want to describe and am confident with the kubectl kind/name format (and the names are short!), I can just enter it.  The two queries to the cluster slow me down here.  And where I want the guided prompts, once I am used to the "hit enter" idiom, the overhead of that is pretty low (though the initial cognitive overhead is significant),

This is an area where I wish VS Code had a richer widget set - if we could have something where we could allow the user to type but still display a list of suggestions, that would solve so many problems.  Oh well!

I guess I am okay with it but maybe we should consider a setting to restore the old behaviour... though that adds more complexity and more documentation effort...

Reply:
I was looking for a way to show the quickpick and also allow users to type something else (a mix between quickpick and inputbox). I guess we can make something similar as i found https://github.com/microsoft/vscode/issues/89601 ... but then i thought we would still have the multiple queries to the cluster so i opted for the setting as you suggested.